File: csharp/perfect-numbers/PerfectNumbers.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;

public enum Classification
{
    Perfect,
    Abundant,
    Deficient
}

public static class PerfectNumbers
{
    public static Classification Classify(int number)
    {
        Classification classification;
        switch (GetAliqutoSum(number))
        {
            case int aliquot when aliquot == number:
                classification = Classification.Perfect;
                break;
            case int aliquot when aliquot > number:
                classification = Classification.Abundant;
                break;
            case int aliquto when aliquto < number:
                classification = Classification.Deficient;
                break;
            default:
                throw new Exception("The factor is invalid");
        }

        return classification;
    }

    private static int GetAliqutoSum(int number) => Enumerable.Range(1, number - 1).Where(i => number % i == 0).Sum();
}

File: python/perfect-numbers/perfect_numbers.py
"""exercise"""

ABUNDANT = "abundant"
DEFICIENT = "deficient"
PERFECT = "perfect"

def classify(number):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """

    # if a number to be classified is less than 1.
    if number < 1:
        raise ValueError("Classification is only possible for positive integers.")

    aliquot_sum = sum([i for i in range(1,number//2+1) if number % i == 0])
    if aliquot_sum == number:
        classification = PERFECT
    elif aliquot_sum < number:
        classification = DEFICIENT
    else:
        classification = ABUNDANT

    return classification
